source/common/config/discovery_grpc_stream.h

```diff
@@ -0,0 +1,133 @@
+#pragma once
+
+#include <functional>
+
+#include "envoy/grpc/async_client.h"
+
+#include "common/common/backoff_strategy.h"
+#include "common/common/token_bucket_impl.h"
+#include "common/config/utility.h"
+
+namespace Envoy {
+namespace Config {
+
+// Oversees communication for gRPC-using xDS: reestablishes the gRPC channel when necessary, and
+// provides rate limiting of requests.
+template <class RequestProto, class ResponseProto>
+class DiscoveryGrpcStream : public Grpc::TypedAsyncStreamCallbacks<ResponseProto>,
+                            public Logger::Loggable<Logger::Id::upstream> {
```

Comment:
I don't think you want to log upstream here, probably config?

Reply:
Thanks, didn't even notice that; done.